Assets/Plugins/iOS/Shopify/BuySerializable.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  BuySerializable.swift
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public protocol BuySerializable {
+    func asJSONString() -> String
+    func asDictionary() -> Dictionary<String, Any>
```

Comment:
It seems more intuitive for a `Serializable` type to vend `Data` rather than a string, in my opinion. We can also make this interface more Swifty:
```swift
typealias JSON = [String: Any]

public protocol Serializable {
    func serializedJson() -> JSON
}

extension Serializable {
    
    func serializedData() throws -> Data {
        return try JSONSerialization.data(withJSONObject: self.serializedJson(), options: .prettyPrinted)
    }

    func serializedString() -> String {
        return String(data: self.serializedData(), encoding: .utf8)
    }
}
```

The two methods in the `extension` should not be declared in the protocol. They are derived from `JSON` and are implemented automatically for any types conforming that are able to provide a `JSON` representation of themselves.